kotest-core/src/commonMain/kotlin/io/kotest/core/runtime/callbacks.kt

```diff
@@ -13,22 +13,41 @@ import io.kotest.core.test.TestResult
 import io.kotest.fp.Try
 import io.kotest.mpp.log
 
+
+fun List<ProjectListener>.resolveName() =
```

Comment:
A better name might be groupByName()

Reply:
grouping is only first step. And why this step is needed. JUnit can't register dynamic test with the same name if you has several equaled listener names you should add some different part to name. This is why it called resolveName or make listeners name to be differrent